lib/solr/indexing/document.rb

```diff
@@ -13,12 +13,11 @@ def add_field(name, value)
       end
 
       # TODO: refactor & optimize this
-      def to_json(_json_context)
-        solr_fields = fields.map do |k, v|
-          solr_field_name = solarize_field(k)
+      def as_json(core:)
```

Comment:
I forgot to ask yesterday, why is this change necessary? Would you be able to revert it by specifying a current core on a lower level of configuration?